ppml/services/kms-utils/docker/Dockerfile

```diff
@@ -1,45 +1,87 @@
-FROM ubuntu:20.04
-
-ARG EHSM_KMS_BRANCH_VERSION=main
-ARG SPARK_JAR_REPO_URL
 ARG HTTP_PROXY_HOST
 ARG HTTP_PROXY_PORT
 ARG HTTPS_PROXY_HOST
 ARG HTTPS_PROXY_PORT
+ARG JDK_VERSION=8u192
+ARG SPARK_VERSION=3.1.2
+
+# stage.1 java & spark
+FROM ubuntu:18.04 AS builder
+
+ARG SPARK_VERSION
+ARG JDK_VERSION
+ARG JDK_URL=your_jdk_url
+ARG SPARK_JAR_REPO_URL
+ENV JDK_VERSION         ${JDK_VERSION}
+ENV SPARK_VERSION       ${SPARK_VERSION}
+
+ENV JAVA_HOME           /opt/jdk${JDK_VERSION}
+ENV SPARK_HOME          /opt/spark-${SPARK_VERSION}
+ENV PATH                ${JAVA_HOME}/bin:${SPARK_HOME}/bin:${PATH}
+
+RUN apt-get update --fix-missing && \
+    env DEBIAN_FRONTEND=noninteractive TZ=Etc/UTC apt-get install -y tzdata apt-utils wget unzip patch zip
+
+# java
+RUN wget $JDK_URL && \
+    gunzip jdk-$JDK_VERSION-linux-x64.tar.gz && \
+    tar -xf jdk-$JDK_VERSION-linux-x64.tar -C /opt && \
+    rm jdk-$JDK_VERSION-linux-x64.tar && \
+    mv /opt/jdk* /opt/jdk$JDK_VERSION && \
+    ln -s /opt/jdk$JDK_VERSION /opt/jdk
+
+# spark
+RUN cd /opt && \
+    wget https://archive.apache.org/dist/spark/spark-${SPARK_VERSION}/spark-${SPARK_VERSION}-bin-hadoop3.2.tgz && \
+    tar -zxvf spark-${SPARK_VERSION}-bin-hadoop3.2.tgz && \
+    mv spark-${SPARK_VERSION}-bin-hadoop3.2 spark-${SPARK_VERSION} && \
+    rm spark-${SPARK_VERSION}-bin-hadoop3.2.tgz && \
+    cp spark-${SPARK_VERSION}/conf/log4j.properties.template spark-${SPARK_VERSION}/conf/log4j.properties && \
+    echo $'\nlog4j.logger.io.netty=ERROR' >> spark-${SPARK_VERSION}/conf/log4j.properties
+
+# spark-encrypt-io.jar
+RUN cd /home && \
+    wget $SPARK_JAR_REPO_URL/spark-encrypt-io-0.2-SNAPSHOT.jar -O spark-encrypt-io.jar
```

Comment:
do we still need this?

Reply:
Yes, the spark-encrypt-io was migrated to BigDL PPML

Reply:
spark-encrypt-io.jar is removed, instead, $BIGDL_HOME/jars/bigdl-ppml-spark_3.1.2-2.1.0-SNAPSHOT.jar is involved to generate keys and encrypt/decrypt.